cloudinit/sources/DataSourceVultr.py

```diff
@@ -0,0 +1,155 @@
+# Author: Eric Benner <ebenner@vultr.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+# Vultr Metadata API:
+# https://www.vultr.com/metadata/
+
+from cloudinit import log as log
+from cloudinit import sources
+from cloudinit import util
+
+import cloudinit.sources.helpers.vultr as vultr
+
+LOG = log.getLogger(__name__)
+BUILTIN_DS_CONFIG = {
+    'url': 'http://169.254.169.254',
+    'retries': 30,
+    'timeout': 2,
+    'wait': 2
+}
+
+
+class DataSourceVultr(sources.DataSource):
+
+    dsname = 'Vultr'
+
+    def __init__(self, sys_cfg, distro, paths):
+        super(DataSourceVultr, self).__init__(sys_cfg, distro, paths)
+        self.ds_cfg = util.mergemanydict([
+            util.get_cfg_by_path(sys_cfg, ["datasource", "Vultr"], {}),
+            BUILTIN_DS_CONFIG])
+        BUILTIN_DS_CONFIG['url'] = self.ds_cfg.get(
+            'url', BUILTIN_DS_CONFIG['url'])
+        BUILTIN_DS_CONFIG['retries'] = self.ds_cfg.get(
+            'retries', BUILTIN_DS_CONFIG['retries'])
+        BUILTIN_DS_CONFIG['timeout'] = self.ds_cfg.get(
+            'timeout', BUILTIN_DS_CONFIG['timeout'])
+        BUILTIN_DS_CONFIG['wait'] = self.ds_cfg.get(
+            'wait', BUILTIN_DS_CONFIG['wait'])
```

Comment:
I'm pretty sure these lines (32-39) are not needed.

```
>>> import cloudinit.util as util
>>> util.mergemanydict([{}, BUILTIN_DS_CONFIG])
{'url': 'http://169.254.169.254', 'retries': 30, 'timeout': 2, 'wait': 2}

>>> util.mergemanydict([{'url': 'http://1.1.1.1/foo'}, BUILTIN_DS_CONFIG])
{'url': 'http://1.1.1.1/foo', 'retries': 30, 'timeout': 2, 'wait': 2}
```

Am I wrong?

Reply:
yeah, i'm pretty sure they are *not* required anymore.  As you're just updating BUILTIN_DS_CONFIG.  but since we do not use that global anywhere else (except the main, which won't have executed *this* code), the lines are not needed.

i think they were left over from when you were using a global more.

Reply:
Indeed, just left over code I missed. Its removed now.